core/services/ocr2/plugins/ocr2vrf/coordinator/block_cache.go

```diff
@@ -0,0 +1,122 @@
+package coordinator
+
+import (
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// blockCache is a caching strucuture that allows items to be stored and then evicted
+// based on an eviction window. In this package, it is being used to track in-flight
+// items the coordinator includes in OCR reports, such that the items can be checked against
+// the cache to avoid double-transmissions.
+type blockCache[T any] struct {
+	blockEvictionWindow int64
+	cacheMu             sync.Mutex
+	cache               map[common.Hash]*cacheItem[T]
+	cleaner             *intervalCacheCleaner[T]
+}
+
+type cacheItem[T any] struct {
+	item        T
+	itemKey     common.Hash
+	blockStored int64
```

Comment:
Similar to the previous comment, is this a block number or a timestamp? If it's the latter, we should use `time.Time` rather than `int64` to make that clear. If the former, `int64` is fine.